docs/development/new-cloud-provider-botanist.md

```diff
@@ -0,0 +1,61 @@
+# Botanist Implementation for New Cloud Provider
+
+This document describes how to extend Botanist to support a new cloud provider. There is a significant number of steps necessary.
+
+## Steps
+
+The implementation consists of several parts:
+
+1. Update the helm chart configurations
+2. Update the Gardener golang code to support the cloud provider.
+
+### Helm charts
+
+1. If your implementation is not in-tree to Kubernetes as a cloud provider, add the cloud-controller-manager (CCM) and cloud-storage-interface (CSI) images to [charts/images.yaml](../../charts/images.yaml). This is _just_ a manifest of all _potential_ images to be used in Gardener. The images are referenced in Gardener by the unique `name` key in the file, and can be used elsewhere in Gardener.
+    * Add the provider-specific CCM with a unique name across Gardener
+    * Add the provider-specific CSI with a unique name across Gardener
+2. Update config files for the seed control plane charts in [charts/seed-controlplane/charts/](../../charts/seed-controlplane/charts/):
+    1. If necessary, update the seed control plane chart for the cloud controller manager in `machine-controller-manager/templates/cloud-controller-manager.yaml`
+    2. If necessary, update the default values for the cloud controller manager in `machine-controller-manager/values.yaml`
+    3. Ensure the cloud provider is listed as an option for the cloud volume plugin in the relevant `if contains` statement in `kube-controller-manager/templates/_helpers.tpl`
+3. Create config files for the seed machines charts in [charts/seed-machines/charts](../../charts/seed-machines/charts)::
+    1. Create a directory for the cloud provider machine class as `<provider>-machineclass/Chart.yaml`
+    2. Create a machine class template for the cloud provider as `<provider>-machineclass/templates/<provider>-machineclass.yaml`
+    3. Create default values for the cloud provider as `<provider>-machineclass/values.yaml`
+4. Create config files for the seed terraformer charts in [charts/seed-terraformer/charts/](../../charts/seed-terraformer/charts/):
+    1. If DNS is available, create the template in `<provider>-dns/templates/_main.tf`
+    2. If DNS is available, create the default values in `<provider>-dns/values.yaml`
+    3. Create common infrastructure chart as `<provider>-infra/Chart.yaml`
+    4. List any binary requirements to `<provider>-infra/requirements.lock`
```

Comment:
Where do you see this file?

Reply:
It _was_ in an earlier version of #626. There must have been some changes. I will remove it.